core/src/main/kotlin/com/linecorp/kotlinjdsl/querydsl/QueryDslImpl.kt

```diff
@@ -45,23 +45,24 @@ import javax.persistence.criteria.JoinType
  *
  * Don't use this directly because it's an <string>INTERNAL</strong> class.
  * It does not support backward compatibility.
+ * This class should be used with the understanding that it is not thread safe and therefore not suitable for parallel processing.
  */
 open class QueryDslImpl<T>(
     private val returnType: Class<T>,
 ) : CriteriaQueryDsl<T>, SubqueryDsl<T>, CriteriaUpdateQueryDsl, CriteriaDeleteQueryDsl {
     private var singleSelectClause: SingleSelectClause<T>? = null
     private var multiSelectClause: MultiSelectClause<T>? = null
     private var fromClause: FromClause<*>? = null
-    private var joins: MutableList<JoinSpec<*>> = mutableListOf()
-    private var wheres: MutableList<PredicateSpec> = mutableListOf()
-    private var groupBys: MutableList<ExpressionSpec<*>> = mutableListOf()
-    private var havings: MutableList<PredicateSpec> = mutableListOf()
-    private var orderBys: MutableList<OrderSpec> = mutableListOf()
+    private var joins: Lazy<MutableList<JoinSpec<*>>> = lazy(LazyThreadSafetyMode.NONE) { mutableListOf() }
+    private var wheres: Lazy<MutableList<PredicateSpec>> = lazy(LazyThreadSafetyMode.NONE) { mutableListOf() }
+    private var groupBys: Lazy<MutableList<ExpressionSpec<*>>> = lazy(LazyThreadSafetyMode.NONE) { mutableListOf() }
+    private var havings: Lazy<MutableList<PredicateSpec>> = lazy(LazyThreadSafetyMode.NONE) { mutableListOf() }
+    private var orderBys: Lazy<MutableList<OrderSpec>> = lazy(LazyThreadSafetyMode.NONE) { mutableListOf() }
```

Comment:
-- english
If you want to save memory resources, I think null use would be better than Lazy use.

Lazy objects also have lambda objects and Lazy wrapper objects internally, so I don't think they will be better than null in that they reduce memory usage.

-- korean
메모리 자원을 아낀다면 Lazy의 사용보다 null의 사용이 더 좋을 것 같습니다.

Lazy 객체 또한 내부적으로 람다 객체 및 Lazy 래퍼 객체를 가지고 있기 때문에 메모리 사용량을 줄인다는 점에서 null 보다는 좋지 않을 것 같아요.

Reply:
Yes. null is best .
However, the code is expected to be very messy, so I set it to lazy.
And I worked on it because I thought that lazy would work cheaper than the object array 10 * 8 + 16 * 2
Wouldn't null be better if the code gets messy?

-- korean
null 이 best 는 맞습니다.
그런데 코드가 매우 지저분해질 것으로 예상되어서 lazy 로 하였습니다.
그리고 object 배열 10 * 8 + 16 * 2 보다는 lazy 가 싸게 먹히지 않을까 싶어서 작업을 해보았던건데요
코드가 지저분해져도 null 이 나을까요?

Reply:
-- english
If you worried about code complexity, it is ok to use lazy delegate

-- korean
네 코드가 지저분해지는 것을 우려하셨다면 lazy가 더 적은 비용을 가져가기 때문에 lazy로 가는 것도 좋겠네요.

Reply:
It seems that it is possible to do it more neatly than I thought even if it is null (orEmpty is the original one)
Please review again.

-- korean
null 로 하고도 생각보다 깔끔하게도 가능한듯 합니다 (orEmpty 는 원래도 되었네요)
다시 리뷰 부탁드립니다.